reportportal/values.yaml

```diff
@@ -5,6 +5,9 @@
 nameOverride: ""
 fullnameOverride: ""
 
+global:
```

Comment:
I don't see anything that refers to those values.

Reply:
**rabbitmq**, **postgresql**, and **minio** dependencies use these global values. If you build the dependencies and expand the archives you will see it in each of their chart values. It's generally good to include **imagePullSecrets** as a global value option so it can be shared with all dependent charts, avoiding redundant configuration.

Reply:
I believe it works for RP dependencies when referred [here (e.g. rabbitmq)](https://github.com/reportportal/kubernetes/blob/5780380529297ac402cf7a475c603ce23d8bf00a/reportportal/values.yaml#L623)
```
rabbitmq:
  global:
    imagePullSecrets: []
```

Template: https://github.com/bitnami/charts/blob/main/bitnami/rabbitmq/templates/_helpers.tpl#L25

Reply:
`global.imagePullSecrets` doesn't resolve your issue. It's only RP value and it doesn't have any relations with the chart dependencies. I don't see profit from this change.

Reply:
I'm not sure I'm understanding you correctly - global values will be passed down to all subcharts as described here: https://helm.sh/docs/chart_template_guide/subcharts_and_globals/#global-chart-values i.e. setting the global **imagePullSecrets** in each of the dependent charts

Reply:
If you set the below in the RP chart:
```
global:
  imagePullSecrets:
  - regcred
```
and template the chart you will see it is getting rendered in any relevant Deployments or StatefulSets for the dependent charts.
See example:
```
 ~ helm template reportportal/ --set ingress.class="gce" --set uat.superadminInitPasswd.password=test --set uat.resources.requests.memory="1Gi" --set serviceapi.resources.requests.cpu="1000m" --set serviceapi.resources.requests.memory="2Gi" --set serviceanalyzer.resources.requests.memory="1Gi" -n reportportal
```

**Sample snippet of rendered dependency:**
```
---
# Source: reportportal/charts/minio/templates/standalone/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: release-name-minio
  namespace: reportportal
  labels:
    app.kubernetes.io/name: minio
    helm.sh/chart: minio-6.7.7
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
spec:
  strategy:
    type: Recreate
  selector:
    matchLabels:
      app.kubernetes.io/name: minio
      app.kubernetes.io/instance: release-name
  template:
    metadata:
      labels:
        app.kubernetes.io/name: minio
        helm.sh/chart: minio-6.7.7
        app.kubernetes.io/instance: release-name
        app.kubernetes.io/managed-by: Helm
    spec:
      imagePullSecrets:
        - name: regcred <----- applied via the global value to the MinIO chart dependency
      serviceAccountName: release-name-minio
```

Reply:
See an example in another Helm chart here: https://github.com/kubeshop/helm-charts/blob/develop/charts/testkube/values.yaml#L5-L18

Reply:
> Since the dependent charts are including a block for
> 
> ```
> global:
>   imagePullSecrets: []
> ```
> 
> then setting this value in the RP values will ensure it is passed down to all of the dependencies.

I got it, however why are there `{{ .Values.global.imagePullSecrets }}` and `{{ .Values.imagePullSecrets }}` in the Values file? Could you leave only `{{ .Values.global.imagePullSecrets }}` and delete `{{ .Values.imagePullSecrets }}` ?

Reply:
When I provide `{{ .Values.global.imagePullSecrets }}` in my Value chart, I expect this value to be used in all my templates, not only sub-charts. Isn't that so?